### PR TITLE
Int: Split SDK into 3 separate python packages

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem

We need 3 separate packages to limit what we install on Server side
https://zapatacomputing.atlassian.net/browse/ORQSDK-1034

# This PR's solution
What was done:
- create 2 new packages
- copy to the new projects: Makefiles, licences changelos readmes and pyproject.toml
- Create fake tests so we can leave Makefiles mostly in the same state.
- Fix orquestra-workflow-runtime API
- Move Secrets from orquestra-sdk to orquestra-workflow-shared
- Fix github workflows to properly run on all 3 projects.
- Add proper dependencies (locally) on SDK and runtime packages.

Whats still TODO as part of this effort:
- Fix ugly soft-dependency. In secrets (shared package) we require client package, which is bad and should be fixed
- Publish 2 new packages on PyPi, and actually add proper dependency from PyPi instead of local build (SDK is not publishable until this is done)
- Cleanup and move tests to appropriate packages. Right now everything is on SDK.
- Cleanup pyproject.tomls, as the dependencies are mostly copied over packages. This should be cleaned up.

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: projects/orquestra-workflow-runtime/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+Please refer to the [Orquestra Workflow SDK docs](https://docs.orquestra.io/docs/core/sdk/).
+
+## Bug Reporting
+
+If you'd like to report a bug/issue please create a [new issue using one of the templates](https://github.com/zapata-engineering/orquestra-sdk/issues).
+
+## Contributing
+
+Please see our [CONTRIBUTING.md](CONTRIBUTING.md) for more information on contributing to Orquestra Workflow SDK.
```
Comment (alexjuda) [inline projects/orquestra-workflow-runtime/README.md:32]: **nitpick**: this file doesn't seem to describe `orquestra-workflow-runtime`. We don't have to be very verbose about it, but no information is better than misleading information:
```suggestion
# Orquestra Workflow Runtime

Executes Orquestra workflows. This library isn't meant to be used directly by external users, please use [orquestra-sdk](https://github.com/zapata-engineering/orquestra-sdk/tree/main/projects/orquestra-sdk) instead.
```

File: projects/orquestra-workflow-runtime/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+[tool.hatch.metadata]
+allow-direct-references = true  # TO BE REMOVED WHEN NO LONGER INSTALLING PACKAGES FROM LOCAL DIRS
+
+[project]
+authors = [
+    { name = "Zapata Computing Inc.", email = "info@zapatacomputing.com" },
+]
+name = "orquestra-workflow-runtime"
+description = "Compose Orquestra workflows using a Python DSL"
```
Comment (alexjuda) [inline projects/orquestra-workflow-runtime/pyproject.toml:13]:
```suggestion
description = "Execute Orquestra workflows"
```

File: projects/orquestra-workflow-runtime/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,239 @@
+[build-system]
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
+[tool.hatch.metadata]
+allow-direct-references = true  # TO BE REMOVED WHEN NO LONGER INSTALLING PACKAGES FROM LOCAL DIRS
+
+[project]
+authors = [
+    { name = "Zapata Computing Inc.", email = "info@zapatacomputing.com" },
```
Comment (alexjuda) [inline projects/orquestra-workflow-runtime/pyproject.toml:10]:
```suggestion
    { name = "Zapata Computing Inc.", email = "info@zapata.ai" },
```

File: projects/orquestra-workflow-runtime/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+    "cloup>=2.0",
+    "inquirer>=3.0",
+    "tabulate",
+    "rich>=13.5",
+    # For automatic login
+    "aiohttp>=3.9.0",
+    # Decoding JWTs
+    "PyJWT>=2.6",
+    # Capture stdout/stderr
+    "wurlitzer>=3.0",
```
Comment (alexjuda) [inline projects/orquestra-workflow-runtime/pyproject.toml:69]: **non-blocking**: please remember to clean up the dependencies before marking this ticket as done.

File: projects/orquestra-workflow-shared/Makefile [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+# human-readable report to stdout.
+show-coverage-text-report:
+	$(PYTHON) -m coverage report --show-missing
+
+BASE_BRANCH := $(if $(GITHUB_BASE_REF),$(GITHUB_BASE_REF),main)
+github-actions-coverage-report:
+	@$(PYTHON) -m coverage report --show-missing
+	@$(PYTHON) -m diff_cover.diff_cover_tool coverage.xml --compare-branch=origin/$(BASE_BRANCH)
+
+github_actions:
```
Comment (alexjuda) [inline projects/orquestra-workflow-shared/Makefile:58]:
```suggestion
github-actions:
```

File: projects/orquestra-workflow-shared/README.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+Please refer to the [Orquestra Workflow SDK docs](https://docs.orquestra.io/docs/core/sdk/).
+
+## Bug Reporting
+
+If you'd like to report a bug/issue please create a [new issue using one of the templates](https://github.com/zapata-engineering/orquestra-sdk/issues).
+
+## Contributing
+
+Please see our [CONTRIBUTING.md](CONTRIBUTING.md) for more information on contributing to Orquestra Workflow SDK.
```
Comment (alexjuda) [inline projects/orquestra-workflow-shared/README.md:32]:
```suggestion
# Orquestra Workflow Shared

This library isn't meant to be used directly by external users. Please use [orquestra-sdk](https://github.com/zapata-engineering/orquestra-sdk/tree/main/projects/orquestra-sdk) instead.

Interfaces and other code we have to share between the user-facing clients and the workflow execution servers. We should strive to keep it as small as possible.
```

File: projects/orquestra-workflow-shared/pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,224 @@
+[build-system]
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
+[project]
+authors = [
+    { name = "Zapata Computing Inc.", email = "info@zapatacomputing.com" },
+]
+name = "orquestra-workflow-shared"
+description = "Compose Orquestra workflows using a Python DSL"
```
Comment (alexjuda) [inline projects/orquestra-workflow-shared/pyproject.toml:10]:
```suggestion
description = "Code shared between Orquestra SDK clients and workflow execution servers"
```

File: projects/orquestra-workflow-shared/pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+    # Decoding JWTs
+    "PyJWT>=2.6",
+    # Capture stdout/stderr
+    "wurlitzer>=3.0",
+    # For dremio client
+    # pyarrow 16.0 crashed ray workers for unknown reason. Crashes were not
+    # reproducable on mac - so carefull with taking that restriction away.
+    "pyarrow>=10.0,<16.0",
+    "pandas>=1.4",
+]
```
Comment (alexjuda) [inline projects/orquestra-workflow-shared/pyproject.toml:71]: **non-blocking**: please remember to clean up the dependencies before marking this ticket as done.

```suggestion
dependencies = [
    # Schema definition
    "pydantic>1.10.7",
    # Pickling library
    "cloudpickle==2.2.1",
    # We should keep `dill` to make sure old workflows can be unpickled.
    "dill==0.3.6",
    # Enhances wrapping of user's functions
    "wrapt",
    # Configuration file locking
    "filelock>=3.3.2",
    # Parse version strings
    "packaging>=21",
    # For different import types in SDK
    "GitPython",
    "pip-api>=0.0.30",
    # Python Standard Library Backports
    "importlib_metadata>=4.0",
    "typing-extensions>=4.1.0",
    # Web requests for external services
    "requests>=2.28",
    # Nice workflow visualizations
    "graphviz",
    # CLI utilities
    "argcomplete",
    # click 8.1.4 introduced type checking errors
    # see https://github.com/pallets/click/issues/2558
    "click>=8.0, != 8.1.4",
    "cloup>=2.0",
    "inquirer>=3.0",
    "tabulate",
    "rich>=13.5",
    # For automatic login
    "aiohttp>=3.9.0",
    # Decoding JWTs
    "PyJWT>=2.6",
    # Capture stdout/stderr
    "wurlitzer>=3.0",
    # For dremio client
    # pyarrow 16.0 crashed ray workers for unknown reason. Crashes were not
    # reproducable on mac - so carefull with taking that restriction away.
    "pyarrow>=10.0,<16.0",
    "pandas>=1.4",
]
```

**Review by alexjuda — CHANGES_REQUESTED** [review]

File: projects/orquestra-workflow-runtime/Makefile [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+# human-readable report to stdout.
+show-coverage-text-report:
+	$(PYTHON) -m coverage report --show-missing
+
+BASE_BRANCH := $(if $(GITHUB_BASE_REF),$(GITHUB_BASE_REF),main)
+github-actions-coverage-report:
+	@$(PYTHON) -m coverage report --show-missing
+	@$(PYTHON) -m diff_cover.diff_cover_tool coverage.xml --compare-branch=origin/$(BASE_BRANCH)
+
+github_actions:
```
[the remark this answers is not in the document]
Comment (SebastianMorawiec) [reply projects/orquestra-workflow-runtime/Makefile:58]: This is copy of the file from original orquestra-sdk, and we have to have the same naming everywhere. This action is called across all the packages from githbu workflow directly. Ill leave it as it is if you dont mind for now.

File: projects/orquestra-workflow-runtime/pyproject.toml (hunk 4 above)
Comment (SebastianMorawiec) [reply projects/orquestra-workflow-runtime/pyproject.toml:69]: Yep, one of TODOs on the list before closing this ticket

File: projects/orquestra-workflow-shared/Makefile (hunk 5 above)
Comment (SebastianMorawiec) [reply projects/orquestra-workflow-shared/Makefile:58]: Same as in previous comment, this has to be consistent across all 3 packages and Ill rather leave it as it is for now

**Review by alexjuda — APPROVED** [review]

**Pull request merged**